File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];

        for (int i = 0; i < queries.length; i++) {
            int start = queries[i][0];
            int end = queries[i][1];
            int k = queries[i][2];

            int[] subArray = Arrays.copyOfRange(arr, start, end + 1);
            Arrays.sort(subArray);

            int result = -1;
            for (int num : subArray) {
                if (num > k) {
                    result = num;
                    break;
                }
            }

            answer[i] = result;
        }

        return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []

    for query in queries:
        start, end, k = query

        sub_array = arr[start:end + 1]
        sub_array.sort()

        result = -1
        for num in sub_array:
            if num > k:
                result = num
                break

        answer.append(result)

    return answer
